File: app/src/main/java/com/kevin/java/annotation/event/OnLongClick.java
package com.kevin.java.annotation.event;

import android.view.View;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
@EventBus(eventType = View.OnLongClickListener.class, eventMethod = "setOnLongClickListener")
public @interface OnLongClick {

    int[] value();
}

File: app/src/main/java/com/kevin/injecttest/annotation/event/OnLongClick.kt
package com.kevin.injecttest.annotation.event

import android.view.View


@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER
)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Event("setOnLongClickListener", View.OnLongClickListener::class, "onLongClick")
annotation class OnLongClick(vararg val ids: Int)